legal-api/src/legal_api/resources/v2/business/business.py

```diff
@@ -167,56 +166,45 @@ def search_businesses():
-                business_json['alternateNames'] = business.get_alternate_names()
-            bus_results.append(business_json)
-
-        draft_results = []
-        for draft_dao in draft_query.all():
-            draft = {
-                'identifier': draft_dao.temp_reg,  # Temporary registration number of the draft entity
-                'legalType': draft_dao.json_legal_type,  # Legal type of the draft entity
-                'draftType': Filing.FILINGS.get(draft_dao.filing_type, {}).get('temporaryCorpTypeCode'),
-                'draftStatus': draft_dao.status
-            }
-
-            if (draft_dao.status == Filing.Status.PAID.value and
-                    draft_dao.effective_date and draft_dao.effective_date > datetime.now(timezone.utc)):
-                draft['effectiveDate'] = draft_dao.effective_date.isoformat()
-
-            if draft_dao.json_nr:
-                draft['nrNumber'] = draft_dao.json_nr  # Name request number, if available
-            # Retrieves the legal name from the filing JSON. Defaults to None if not found.
-            draft['legalName'] = (draft_dao.filing_json.get('filing', {})
-                                  .get(draft_dao.filing_type, {})
-                                  .get('nameRequest', {})
-                                  .get('legalName'))
-
-            if draft['legalName'] is None:
-                # Fallback to a generic legal name based on the legal type if no specific legal name is found
-                draft['legalName'] = (Business.BUSINESSES
-                                      .get(draft_dao.json_legal_type, {})
-                                      .get('numberedDescription'))
-            draft_results.append(draft)
-
-        return jsonify({'businessEntities': bus_results, 'draftEntities': draft_results}), HTTPStatus.OK
+        for identifier in identifiers:
+            if identifier.startswith('T'):
+                temp_identifiers.append(identifier)
+            else:
+                business_identifiers.append(identifier)
+            
+        bus_results = BusinessSearchService.get_search_filtered_businesses_results(
+            business_json=json_input,
+            identifiers=business_identifiers,
+            search_filter_name=search_filter_name,
+            search_filter_type=search_filter_type,
+            search_filter_status=search_filter_status)
+
+        draft_results = BusinessSearchService.get_search_filtered_filings_results(
+            business_json=json_input,
+            identifiers=temp_identifiers,
+            search_filter_name=search_filter_name,
+            search_filter_type=search_filter_type,
+            search_filter_status=search_filter_status)
+
+        return jsonify({'businessEntities': BusinessSearchService.paginate(bus_results, page, limit), 'draftEntities': BusinessSearchService.paginate(draft_results, page, limit)}), HTTPStatus.OK
```

Comment:
You're fetching and then paginating after? why not just paginate at the DAO level?